docker/hisat2/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM ubuntu:17.04
+MAINTAINER Jishu Xu <jishuxu@broadinstitute.org>
+
+LABEL software="HISAT2"
+LABEL version="2-2.1.0"
+LABEL description="RNA-seq aligner"
+LABEL website="https://ccb.jhu.edu/software/hisat2/index.shtml"
+
+RUN mkdir -p /opt/tools/
+WORKDIR /opt/tools
+
+RUN \
+  apt update && \ 
+  apt install -y liblzma-dev libbz2-dev cmake automake curl libboost-all-dev wget build-essential gcc-multilib zlib1g-dev libxml2-dev libncurses5-dev zip unzip samtools
+  
+RUN \
+  wget -c ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.1.0-source.zip && \
+  unzip hisat2-2.1.0-source.zip && \
+  cd hisat2-2.1.0 && \
+  make && \
+  cp hisat2* /usr/local/bin
```

Comment:
could probably remove anything in the source directory that isn't copied and the compressed zipfile to reduce space.

Reply:
probably not, there are a bunch of useful scripts come with HISAT2 source code. we will need some of them in pipeline.